gateway-contracts/contracts/interfaces/IInputVerification.sol

```diff
@@ -27,10 +29,34 @@ interface IInputVerification {
     /**
      * @notice Emitted once a correct ZK Proof verification is completed.
      * @param zkProofId The ID of the ZK Proof.
+     * @param coprocessorContextId The ID of the coprocessor context.
      * @param ctHandles The coprocessor's computed ciphertext handles.
      * @param signatures The coprocessor's signature.
      */
-    event VerifyProofResponse(uint256 indexed zkProofId, bytes32[] ctHandles, bytes[] signatures);
+    event VerifyProofResponse(
+        uint256 indexed zkProofId,
+        uint256 indexed coprocessorContextId,
+        bytes32[] ctHandles,
+        bytes[] signatures
+    );
+
+    /**
+     * @notice Error indicating that the coprocessor context is no longer valid for verifying the ZK Proof.
+     * A context is valid if it is active or suspended.
+     * @param zkProofId The ID of the ZK Proof.
+     * @param contextId The context ID of the coprocessor.
+     * @param contextStatus The status of the coprocessor context.
+     */
+    error InvalidCoprocessorContextProofVerification(uint256 zkProofId, uint256 contextId, ContextStatus contextStatus);
+
+    /**
+     * @notice Error indicating that the coprocessor context is no longer valid for rejecting the ZK Proof.
+     * A context is valid if it is active or suspended.
+     * @param zkProofId The ID of the ZK Proof.
+     * @param contextId The context ID of the coprocessor.
+     * @param contextStatus The status of the coprocessor context.
+     */
+    error InvalidCoprocessorContextProofRejection(uint256 zkProofId, uint256 contextId, ContextStatus contextStatus);
```

Comment:
@isaacdecoded I guess we need to consider these as non-errors and not retry them in the txn-sender as they won't ever succeed, right?